### PR TITLE
✨ Add HasOwnerReference

### PR DESCRIPTION
**Pull request by sivchari**
I added HasOwnerReference API. This API check if owners contains specific owner.

### CODE REVIEW

File: pkg/controller/controllerutil/controllerutil.go [hunk 1]
```diff
@@ -181,6 +181,18 @@ func HasControllerReference(object metav1.Object) bool {
 	return false
 }
 
+// HasOwnerReference returns true if the owners list contains an owner reference
+// that matches the object's group, kind, and name.
+func HasOwnerReference(owners []metav1.OwnerReference, obj client.Object) bool {
+	gvk := obj.GetObjectKind().GroupVersionKind()
```
Comment (troy0820) [inline pkg/controller/controllerutil/controllerutil.go:187]: Can we use the `indexOwnerRefs` function?

Comment (troy0820) [inline pkg/controller/controllerutil/controllerutil.go:187]: There is a function `apiutil.GVKForObject()` that would be useful here.  You would have to interface assert this to a runtime object.

Comment (FillZpp) [reply pkg/controller/controllerutil/controllerutil.go:187]: +1, `TypeMeta` may not always exist in the object, even c-r has made some workaround for it
https://github.com/kubernetes/kubernetes/issues/3030
https://github.com/kubernetes/kubernetes/issues/80609

Comment (sivchari) [reply pkg/controller/controllerutil/controllerutil.go]: I rewrote like [this](https://github.com/kubernetes-sigs/controller-runtime/pull/2882/commits/364384a51c8a2ee2c9830b850cf622d7d8d7f0cb).
Is it right, I want the feedback, thanks.

File: pkg/controller/controllerutil/controllerutil.go [hunk 2]
```diff
@@ -181,6 +181,21 @@ func HasControllerReference(object metav1.Object) bool {
 	return false
 }
 
+// HasOwnerReference returns true if the owners list contains an owner reference
+// that matches the object's group, kind, and name.
+func HasOwnerReference(ownerRefs []metav1.OwnerReference, obj client.Object, scheme *runtime.Scheme) bool {
+	gvk, err := apiutil.GVKForObject(obj, scheme)
+	if err != nil {
+		return false
```
Comment (alvaroaleman) [inline pkg/controller/controllerutil/controllerutil.go:189]: Can we return an error here, please?

File: pkg/controller/controllerutil/controllerutil.go [hunk 3]
```diff
@@ -181,6 +181,21 @@ func HasControllerReference(object metav1.Object) bool {
...
+	}
+	idx := indexOwnerRef(ownerRefs, metav1.OwnerReference{
+		APIVersion: gvk.GroupVersion().String(),
+		Name:       obj.GetName(),
+		Kind:       gvk.Kind,
+	})
+	return idx != -1
+}
```
Comment (alvaroaleman) [inline pkg/controller/controllerutil/controllerutil.go:197]: Please add tests

**Review by vincepri — COMMENTED** [review]
/lgtm
/assign @alvaroaleman

**Comment by vincepri** [comment]
please squash commits

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**